src/main/java/com/microsoft/jenkins/containeragents/builders/AciDeploymentTemplateBuilder.java

```diff
@@ -0,0 +1,341 @@
+
+public final class AciDeploymentTemplateBuilder {
+
+    private static final String DEPLOY_TEMPLATE_FILENAME
+            = "/com/microsoft/jenkins/containeragents/aci/deployTemplate.json";
+    private static final String NETWORK_PROFILE_SNIPPET_FILENAME
+            = "/com/microsoft/jenkins/containeragents/aci/networkProfileSnippet.json";
+    private AciDeploymentTemplateBuilder() {
+        //
+    }
+
+    @NotNull
+    public static AciDeploymentTemplate buildDeploymentTemplate(AciCloud cloud, AciContainerTemplate template,
+                                                                AciAgent agent) throws IOException {
+        try (InputStream stream = AciService.class.getResourceAsStream(DEPLOY_TEMPLATE_FILENAME)) {
+            AciPrivateIpAddress privateIpAddress = template.getPrivateIpAddress();
+
+            final ObjectMapper mapper = new ObjectMapper();
+            final JsonNode tmp = mapper.readTree(stream);
+            final ObjectNode parameters = mapper.createObjectNode();
+
+            ObjectNode variables = (ObjectNode) tmp.get("variables");
+            variables.put("containerName", agent.getNodeName());
+            variables.put("containerImage", template.getImage());
+            variables.put("osType", template.getOsType());
+            variables.put("ipType", mapIpType(privateIpAddress));
+            if (privateIpAddress != null) {
+                variables.put("vnetName", privateIpAddress.getVnet());
+                variables.put("subnetName", privateIpAddress.getSubnet());
+            }
+            variables.put("cpu", template.getCpu());
+            variables.put("memory", template.getMemory());
+            variables.put("jenkinsInstance",
+                    Jenkins.get().getLegacyInstanceId());
+
+            addLogAnalytics(tmp, parameters, mapper, cloud);
+            addCommandNode(tmp, template.getCommand(), agent);
+
+            for (AciPort port : template.getPorts()) {
+                if (StringUtils.isBlank(port.getPort())) {
+                    continue;
+                }
+                addPortNode(tmp, mapper, port.getPort());
+            }
+            if (template.getLaunchMethodType().equals(Constants.LAUNCH_METHOD_SSH)) {
+                addPortNode(tmp, mapper, String.valueOf(template.getSshPort()));
+            }
+
+            addEnvNode(tmp, mapper, template.getEnvVars());
+
+            for (DockerRegistryEndpoint registryEndpoint : template.getPrivateRegistryCredentials()) {
+                addImageRegistryCredentialNode(tmp, mapper, registryEndpoint);
+            }
+
+            for (AzureFileVolume volume : template.getVolumes()) {
+                if (StringUtils.isBlank(volume.getMountPath())
+                        || StringUtils.isBlank(volume.getShareName())
+                        || StringUtils.isBlank(volume.getCredentialsId())) {
+                    continue;
+                }
+                addAzureFileVolumeNode(tmp, mapper, volume);
```

Comment:
formatting may be off but simplified again
```suggestion
                if (StringUtils.isNotBlank(volume.getMountPath()) &&
                StringUtils.isNotBlank(volume.getShareName()) &&
                StringUtils.isNotBlank(volume.getCredentialsId())) {
                    addAzureFileVolumeNode(tmp, mapper, volume);
                }
```